compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1617,6 +1627,28 @@ impl<'tcx> InferCtxt<'tcx> {
         tcx.const_eval_resolve_for_typeck(param_env_erased, unevaluated, span)
     }
 
+    /// The returned function is used in a fast path. If it returns `true` the variable is unchanged,
+    /// `false` indicates that the status is unknown.
+    #[inline]
+    pub fn is_ty_infer_var_unchanged<'a>(
```

Comment:
`is_ty_infer_var_definitely_unchanged` would communicate the unchanged/unknown dichotomy.